src/Microsoft.AspNetCore.Blazor.Razor.Extensions/EliminateMethodBodyPass.cs

```diff
@@ -25,7 +25,42 @@ protected override void ExecuteCore(RazorCodeDocument codeDocument, DocumentInte
             }
 
             var method = documentNode.FindPrimaryMethod();
+            if (method == null)
+            {
+                return;
+            }
+
             method.Children.Clear();
+
+            // After we clear all of the method body there might be some unused fields, which can be
+            // blocking if compiling with warnings as errors. Suppress this warning so that it doesn't
+            // get annoying in VS.
+            documentNode.Children.Insert(documentNode.Children.IndexOf(documentNode.FindPrimaryNamespace()), new CSharpCodeIntermediateNode()
+            {
+                Children =
+                {
+                    // Field is assigned but never used
+                    new IntermediateToken()
+                    {
+                        Content = "#pragma warning disable 0414" + Environment.NewLine,
+                        Kind = TokenKind.CSharp,
+                    },
+
+                    // Field is never assigned
+                    new IntermediateToken()
+                    {
+                        Content = "#pragma warning disable 0649" + Environment.NewLine,
+                        Kind = TokenKind.CSharp,
+                    },
+
+                    // Field is never used
+                    new IntermediateToken()
+                    {
+                        Content = "#pragma warning disable 0169" + Environment.NewLine,
```

Comment:
We don't need corresponding `restore` pragmas because these are all separate files now, is that right?

Reply:
The restore parameters are really just about being nice, and we try to put them **around **specific things when we do pragmas.  In this case the code that triggers the warnings could be anywhere in the file, so we can't really scope it down, it has to surround the whole class.

Also these case of these things, they are only for the temp-compile, so if you have code that would cause these warnings legitimately you will still get them when you build.

I also considered turning off warnings globally when invoking CSC - that may be a better approach, but it would have taken me more work to figure out than this took to code up.